dg.go

```diff
@@ -23,21 +24,34 @@ type directedGraph[NodeType any] struct {
     connectionsToNode   map[string]map[string]struct{}
 }
 
+var errorPathRegex, _ = regexp.Compile("error|crashed|failed|deploy_failed")
+
 func (d *directedGraph[NodeType]) Mermaid() string {
-    result := []string{}
-    result = append(result, "flowchart TD")
-    result = append(result, "subgraph input")
-    for source := range d.connectionsFromNode {
-	if strings.HasPrefix(source, "input") {
-            result = append(result, source)
-	}
+    result := []string{
+        "%% Mermaid markdown workflow",
+        "flowchart LR",
+        "%% Success path",
     }
-    result = append(result, "end")
+
+    errorPath := []string{"%% Error path"}
+
     for source, d := range d.connectionsFromNode {
         for destination := range d {
-            result = append(result, fmt.Sprintf("%s-->%s", source, destination))
+            destinationNodes := strings.Split(destination, ".")
+            isErrorPath := errorPathRegex.MatchString(
+                destinationNodes[len(destinationNodes)-1],
+            )
+            if (isErrorPath) {
+                errorPath = append(errorPath, fmt.Sprintf("%s-->%s", source, destination))
+            } else {
+                result = append(result, fmt.Sprintf("%s-->%s", source, destination))
+            }
```

Comment:
Since the output pattern is the same regardless of whether it's a success or error path, we should either use a constant for the format string or just factor out the formatting operation:
```go
            line := fmt.Sprintf("%s-->%s", source, destination)
            if (isErrorPath) {
                errorPath = append(errorPath, line)
            } else {
                result = append(result, line)
            }
```